### ISSUE TITLE
Custom icon with new icon manager does not display on insert initially.

### ISSUE THREAD

**Issue by mondrey** [elementor/elementor#8951]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description

Using the new icon manager.
Doing everything according to https://developers.elementor.com/new-control-icons/

I have an iconset loaded using the filter 'elementor/icons_manager/additional_tabs'
The json data is stored as a JS file as elementor does.

Sample json. Shortened data for easier reading. Last icon data which is in test

```json
{
  "icons": [
      "alert",
      "alert-circled",
      "android-add",
      "android-add-circle",
      "android-alarm-clock",
      "android-alert",
      "waterdrop",
      "wifi",
      "wineglass",
      "woman",
      "wrench",
      "xbox",
      "ios-bell",
      "ios-bell-outline"
  ]
}
```

```php
add_filter( 'elementor/icons_manager/additional_tabs', 'customicons_add_ionicons_to_icon_manager');

function customicons_add_ionicons_to_icon_manager( $settings ) {
    $json_data = plugin_dir_url( __FILE__ ) . '/assets/fonts/ionicons/js/ionicons.js';
    $settings['ionicons'] = [
        'name'          => 'ion',
        'label'         => esc_html__( 'Ion icons', 'locale' ),
        'url'           => '',
        'enqueue'       => '',
        'prefix'        => 'ion-',
        'displayPrefix' => 'ion',
        'labelIcon'     => 'ion-android-add',
        'ver'           => '1.0',
        'fetchJson'     => $json_data,
    ];

    return $settings;
}
```

Here's how the control is registered

```php
$this->add_control(
	'new_icon',
	[
		'label' => __( 'Choose Icon', 'locale' ),
		'type' => \Elementor\Controls_Manager::ICONS,
		'fa4compatibility' => 'icon',
		'default' => [
			'value' => 'fas fa-star',
			'library' => 'fa-solid',
		],
	]
);
```

the icon ID holds the data `ion-ios-bell-outline`. That's on a previous save with old icon control

## Steps to reproduce

After changing to the new icon manager

a. Clicking on the icon which is stored in ID `icon` displays the new icon library.
b. It also displays the custom icon list. No problem.
c. The `__fa4_migrated` is created.
d. But it also sets the `new_icon` ID as

```php
[new_icon] => Array
        (
            [value] => fasion-ios-bell-outline
            [library] => fa-regular
        )
```

There's the prefix `fas` added which displays an empty icon area.
And the same in icon manager preview is empty as well.
I tracked the `fas` prefix to a string replace in editor.js

```javascript
value: 'fas' + fa4Value.replace('fa ', ' '),
```

From here onwards:

e. Selecting a new icon from the new icon manager does not update the icon.
f. It also does not reflect the change in `new_icon` ID if print_r is used on $settings
g. But it does update the `var_Elementor` data with the new icon.
h. If a new Elementor widget is selected before choosing the icon widget again and changing the icon, then it does display fine.

h. Updating and viewing the page shows the new icon on front end theme.

i.  Updating and reloading the browser Elementor editor page does not show the new icon. But selecting it previews the new icon in the icon manager preview. And choosing another icon displays it in widget preview.

I've gone through the methods but could not figure out if i'm doing anything wrong. There are also no results on this particular issue anywhere.

Sorry if i missed including any information.

Thanks,
Mondrey

## Isolating the problem

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [ ] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>
Mac OS Mojave 10.14.6
Firefox 68.0.2 (64-bit)
</summary>
```
<!-- For bug reports, let us know about your system environment: Copy and paste the system info report from Elementor => System info, and paste it here or in http://pastebin.com/ -->
```
</details>

**Comment by bainternet** [comment]
@mondrey 

Elementor assumes and only migrates FA4 to FA5 icons out of the box, if you want to add your own library to the migration, you need to you mapping to the migration dictionary in Elementor localStorage under the `fa4Tofa5Mapping` sub-key or catch the request for the dictionary and add your icon-set mapping to it.

**Comment by lucasmaxi** [comment]
Hello everyone! I have a problem and I can't find a solution in PT-BR, I've been hunting for a few days and all so far that I tried, they didn't solve.

My menu icons are in a rectangle, YES is selected in the settings (Load support for Font Awesome 4), but the site insists on being without an icon. The icons I can use without problem inside the Elementor, however when I create a submenu, the icon does not load on it. Does anyone have a possible solution?

Thank you very much in advance

Note: the written English must be a little bad to read, I used google translate, sorry for the confusion and thanks for understanding.

**Comment by lucasmaxi** [comment]
Hello everyone! I have a problem and I can't find a solution in PT-BR, I've been hunting for a few days and all so far that I tried, they didn't solve.

My menu icons are in a rectangle, YES is selected in the settings (Load support for Font Awesome 4), but the site insists on being without an icon. The icons I can use without problem inside the Elementor, however when I create a submenu, the icon does not load on it. Does anyone have a possible solution?

Thank you very much in advance

Note: the written English must be a little bad to read, I used google translate, sorry for the confusion and thanks for understanding.
